### PR TITLE
feat(logging): mask sensitive data in record extra and the legacy MonologAdapter

### PR DESCRIPTION
**Pull request by Drago37**
## Description

Extends the `#[Sensitive]` payload masking so it also covers a Monolog record's `extra` bag and the legacy `MonologAdapter`, and consolidates the new-kernel monolog service config.

`SanitizingProcessor` previously masked only `context`. Two gaps are closed:

- **`extra` is now sanitised.** `WebProcessor` stores the request URI (query string included) in `extra.url`; sensitive query parameters are now redacted — on the new-kernel pipeline too. `extra` is masked with keyword-key masking **off**: its keys are set by platform processors (`WebProcessor`, `TokenProcessor`, …), not by callers, so they stay readable for auditing (e.g. `extra.token` is `TokenProcessor`'s audit descriptor of the authenticated user — authenticated flag, roles, identifier — not a credential). Only sensitive values inside URL query strings are redacted there (best-effort, query component only).
- **`MonologAdapter` now runs the `SanitizingProcessor`** (it previously pushed only `ExceptionFormatterProcessor` + `WebProcessor` + `UidProcessor`), so logs emitted through the legacy adapter are masked too.

### Processor ordering

The sanitizer must run **after** `WebProcessor` (which fills `extra.url`). The MonologBundle ignores the `priority` tag for processors and Monolog applies them LIFO, so ordering is controlled by registration position: the sanitizer is registered **first** (in `config.new/services/monolog.php`, and pushed first by `MonologAdapter`) so it executes last.

### Monolog config consolidation

The new-kernel monolog processors were defined in **both** `shared.php` (global) and `monolog.php` (channel-scoped) — a duplication where `shared.php` silently won by load order, leaving `monolog.php`'s channel scoping dead and broken (it referenced the non-existent `bus`/`request` channels). This PR makes `monolog.php` the single home (global processors + formatter + resetter + sanitizer) and reduces `shared.php` to the `App\Shared\` autoloader, matching its siblings. Runtime is unchanged.

## Changes

- `PayloadSanitizer`: URL-query-aware masking (params whose name matches `SensitiveKeywordDenylist`, redacted in place); a `maskKeywordKeys` flag; public `sanitize()` split from a private recursive `walk()` so callers don't thread the internal `$depth`.
- `SanitizingProcessor`: sanitises `context` (full) and `extra` (URL-query only); registered explicitly (attribute removed) to control order.
- `SensitiveKeywordDenylist`: added secret-name aliases (`apikey`, `api-key`, `passwd`, `pwd`, `access_key`, `bearer`, `signature`, `session_id`, `sessionid`).
- `config.new/services/{monolog,shared}.php`: consolidation (single home + pure autoloader).
- `MonologAdapter`: wires the sanitizer (pushed first → runs last).
- `doc/php/logging.md`: documents the `extra`/URL coverage and ordering, scoped honestly (query component only).

## Tests

- `PayloadSanitizerTest`: URL query masking, the `maskKeywordKeys` flag through recursion, the `Stringable` path, query edge cases, case-insensitive / percent-decoded param names, the new aliases, and pinned out-of-scope limitations (secret in path / nested value).
- `SanitizingProcessorTest`: `extra.url` query secret redacted while audit fields (`extra.token`, `extra.ip`) stay intact.
- `MonologAdapterTest`: asserts the sanitizer runs last (after `WebProcessor`).
- All green; `rector:diff` + `cs:diff` clean.

## Verification

Both pipelines verified in a docker environment: the new-kernel container (boot + `getProcessors()` order → sanitizer last on every channel) and the legacy `MonologAdapter` (real log → `extra.url` `token=***`, `context` masked, audit fields preserved).

Fixes: [MON-201784](https://centreon.atlassian.net/browse/MON-201784)

[MON-201784]: https://centreon.atlassian.net/browse/MON-201784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: centreon/config.new/services/monolog.php [hunk 1]
```diff
@@ -34,25 +36,30 @@
...
-        ->tag('monolog.processor', ['channel' => 'request'])
-        ->tag('monolog.processor', ['channel' => 'app']);
+        ->tag('monolog.processor');
 
     $services->set('monolog.processor.token', TokenProcessor::class)
         ->arg('$tokenStorage', service('security.token_storage'))
-        ->tag('monolog.processor', ['channel' => 'bus'])
-        ->tag('monolog.processor', ['channel' => 'request'])
-        ->tag('monolog.processor', ['channel' => 'app']);
+        ->tag('monolog.processor');
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/config.new/services/monolog.php:62]: Checked against the pinned bundle. In `symfony/monolog-bundle` **v3.10.0** (the version vendored here), `AddProcessorsPass::process()` collects `monolog.processor` services via `findTaggedServiceIds()` and pushes them in definition order — it only reads the `channel` / `handler` / `method` tag attributes, never a `priority` one (see `vendor/symfony/monolog-bundle/DependencyInjection/Compiler/AddProcessorsPass.php:34`). Processor `priority` support landed after 3.10.0, so on this version `priority: 100` would be silently ignored and would make the comment inaccurate.

Ordering is therefore controlled by registration position: the sanitizer is registered first → pushed first → runs **last** under Monolog's LIFO. I verified this at runtime via `getProcessors()` on the channel loggers (`SanitizingProcessor` is last, after `WebProcessor`). Keeping the current arrangement and comment; will revisit if the bundle is upgraded to a version that honours processor `priority`.

**Review by asalhi-ext — APPROVED** [review]

**Review by selfakiri — APPROVED** [review]
